code/game/objects/items/storage/fancy.dm

```diff
@@ -293,22 +293,24 @@
 /obj/item/storage/fancy/cigarettes/cigars/update_icon()
 	cut_overlays()
 	if(fancy_open)
-		add_overlay("[icon_state]_open")
+		icon_state = "[initial(icon_state)]_open"
 		var/mutable_appearance/cigar_overlay = mutable_appearance(icon, icon_type)
 		for(var/c = contents.len, c >= 1, c--)
-			cigar_overlay.pixel_x = 4 * (c - 1)
+			cigar_overlay.pixel_x = 3 * (c - 1)

```

Comment:
@JStheguy if you want to take a pop at #15492, all you have to do is make a sprite with all the cigars in the case, then replace this all with 

```
if(contents.len)
	var/mutable_appearance/cigar_overlay = mutable_appearance(icon, "icon_type_[contents.len]")
	add_overlay(cigar_overlay)
```